### PR TITLE
001 employee

### PR DESCRIPTION
**Pull request by lexseaton**

### CODE REVIEW

File: src/main/java/org/example/controllers/EmployeeController.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.xml.transform.Result;
+import java.awt.*;
+import java.sql.SQLException;
+
+@Api("Delivery Employee API")
```
Comment (bbell-kainos) [inline src/main/java/org/example/controllers/EmployeeController.java]: Should this API name not match what the class is called?

File: src/main/java/org/example/controllers/EmployeeController.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    public EmployeeController(DeliveryEmployeeService deliveryEmployeeService) {
+        this.deliveryEmployeeService = deliveryEmployeeService;
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getEmployees() {
+
+        return Response.ok().entity("List of Employees").build();
```
Comment (bbell-kainos) [inline src/main/java/org/example/controllers/EmployeeController.java]: Is this endpoint needed in the requirements? 🤔

File: src/main/java/org/example/daos/DeliveryEmployeeDao.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class DeliveryEmployeeDao {
+
+    public int createDeliveryEmployee(DeliveryEmployeeRequest deliveryEmployee) throws SQLException {
+
+        Connection c = DatabaseConnector.getConnection();
+
+        String insertStatement = "INSERT INTO employee (name, salary, bankNumber, nationalInsuranceNo) VALUES (?,?,?,?);";
```
Comment (bbell-kainos) [inline src/main/java/org/example/daos/DeliveryEmployeeDao.java]: How can we specify if an employee is a delivery employee? Should there be measures to ensure that?

**Review by bbell-kainos — CHANGES_REQUESTED** [review]

File: src/main/java/org/example/controllers/EmployeeController.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response createDeliveryEmployee(DeliveryEmployeeRequest deliveryEmployeeRequest) {
+
+        try {
+            return Response
+                    .status(Response.Status.CREATED)
+                    .entity(deliveryEmployeeService.createDeliveryEmployee(deliveryEmployeeRequest))
+                    .build();
+        } catch (SQLException e){
```
Comment (shaunganley) [inline src/main/java/org/example/controllers/EmployeeController.java:39]: what happens if the request fails validation?

File: src/main/java/org/example/TestApplication.java [hunk 5]
```diff
@@ -28,10 +31,10 @@ protected SwaggerBundleConfiguration getSwaggerBundleConfiguration(
         });
     }
     @Override
-    public void run(final TestConfiguration configuration,
-                    final Environment environment) {
-        environment.jersey()
-                .register(new TestController(new TestService(new TestDao())));
+    public void run(final TestConfiguration configuration, final Environment environment) {
+        environment.jersey().register(new TestController(new TestService(new TestDao())));
```
Comment (shaunganley) [inline src/main/java/org/example/TestApplication.java:35]: test controller not needed anymore?

File: src/main/resources/db/migration/V1__create_employee_table.sql [hunk 6]
```diff
@@ -0,0 +1,19 @@
+CREATE TABLE employee (
```
Comment (shaunganley) [inline src/main/resources/db/migration/V1__create_employee_table.sql:1]: how are we differentiating sales and delivery employees?

File: src/main/java/org/example/services/DeliveryEmployeeService.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+    private DeliveryEmployeeDao delEmpDao;
+
+    public DeliveryEmployeeService(DeliveryEmployeeDao delEmpDao) {
+        this.delEmpDao = delEmpDao;
+    }
+
+    public int createDeliveryEmployee(DeliveryEmployeeRequest deliveryEmployeeRequest) throws SQLException {
+
+        int id  = delEmpDao.createDeliveryEmployee(deliveryEmployeeRequest);
```
Comment (shaunganley) [inline src/main/java/org/example/services/DeliveryEmployeeService.java]: add some validation before sending to db

**Review by shaunganley — CHANGES_REQUESTED** [review]

**Review by shaunganley — CHANGES_REQUESTED** [review]
you need to throw an exception when validation fails otherwise your code will continue as if everything is good

**Review by shaunganley — APPROVED** [review]

**Pull request merged**